packages/common/src/ui/components/navigation/NavigationPrompt/NavigationPrompt.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from 'react';
+// import { Prompt } from 'react-router-dom';
+
+export interface NavigationPromptProps {
+  isUnsaved: boolean;
+}
+
+export const NavigationPrompt: React.FC<NavigationPromptProps> = ({
+  isUnsaved,
+}) => {
+  const promptUser = (e: BeforeUnloadEvent) => {
+    // Cancel the event
+    e.preventDefault(); // If you prevent default behavior in Mozilla Firefox prompt will always be shown
+    // Chrome requires returnValue to be set
+    e.returnValue = '';
+  };
+
+  useEffect(() => {
+    if (isUnsaved) {
+      window.addEventListener('beforeunload', promptUser, { capture: true });
+    } else {
+      window.removeEventListener('beforeunload', promptUser, { capture: true });
+    }
+  }, [isUnsaved]);
+
+  return <></>;
+  // Prompt, usePrompt and useBlocker are not currently implemented in the current v6 release
```

Comment:
it is in v5 - they said they did not include in v6 in order to get the release out.
I don't think it matters - in our use case, you cannot navigate without closing the modal which cancels changes. So the window event is the only thing which can catch the navigation anyway.